obspy/core/util/testing.py

```diff
@@ -113,23 +114,113 @@ def add_doctests(testsuite, module_name):
                 pass
 
 
-def checkForMatplotlibCompareImages():
-    try:
-        # trying to stay inside 80 char line
-        import matplotlib.testing.compare as _compare
-        compare_images = _compare.compare_images  # NOQA @UnusedVariable
-    except:
-        return False
-    # matplotlib's (< 1.2) compare_images() uses PIL internally
-    if getMatplotlibVersion() < [1, 2, 0]:
-        try:
-            import PIL  # NOQA @UnusedImport
-        except ImportError:
-            return False
-    return True
-
-
-HAS_COMPARE_IMAGE = checkForMatplotlibCompareImages()
+def compare_images(expected, actual, tol):
+    """
+    Custom version of :func:`matplotlib.testing.compare.compare_images`.
+    This enable ObsPy to have the same image comparison metrics across
+    matplotlib versions. Futhermore nose is no longer a test dependency of
+    ObsPy.
+
+    Only works with png files in contrast to the matplotlib version.
+
+    Compare two "image" files checking differences within a tolerance.
+
+    Parameters
+    ----------
+    expected : str
+        The filename of the expected image.
+    actual :str
+        The filename of the actual image.
+    tol : float
+        The tolerance (a color value difference, where 255 is the
+        maximal difference).  The test fails if the average pixel
+        difference is greater than this value.
+
+    Example
+    -------
+    img1 = "./baseline/plot.png"
+    img2 = "./output/plot.png"
+    compare_images( img1, img2, 0.001 ):
+    """
+    from matplotlib import _png
```

Comment:
I actually ended up writing my own png writing function to work across matplotlib function...I'll push tomorrow.